securesystemslib/formats.py

```diff
@@ -42,7 +42,7 @@ def _canonical_string_encoder(string):
       A string with the canonical-encoded 'string' embedded.
     """
 
-    string = '"%s"' % string.replace("\\", "\\\\").replace('"', '\\"')
+    string = '"{}"'.format(string.replace("\\", "\\\\").replace('"', '\\"'))
```

Comment:
this seems to be the important line in this patch -- everythings else is error messages or otherwise non-critical. This seems correct to me.

I assume the reason for using format() here is the "quote-quoting" that would be tricky with f-strings?

Reply:
Yes the issue is with the `.replace('"', '\\"')`. I guess we can turn it into an f-string by escaping these quotes, using format might be more readable in the end though.

Reply:
I am mistaken, backslashes are not allowed inside the f-string expression. Keeping it as is will be the most readable solution and is accepted by the linter.